core/src/main/scala/chisel3/Mem.scala

```diff
@@ -60,56 +60,57 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends H
     */
   def apply(x: Int): T = macro SourceInfoTransform.xArg
 
+  def do_apply_impl(idx: UInt, clock: Clock)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = {
```

Comment:
```suggestion
  private def do_apply_impl(idx: UInt, clock: Clock, warn: Boolean)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = {
```

Reply:
Should be private and we need the ability to suppress the warning

Reply:
I think I need protected here since I need to access it from the SyncReadMem subclass

Reply:
> I think I need protected here since I need to access it from the SyncReadMem subclass

Ah yes, correct, but definitely protected instead of public.